File: 卡码/代码随想录/1/06-二叉树/Js/04-二叉搜索树属性/LeetCode_0538.js
//538. 把二叉搜索树转换为累加树
import TreeNode from "../TreeNode.js"

const convertBST = function(root) {
  let pre=0;
  const traversal=root=>{
    if(!root) return;
    traversal(root.right);
    root.val+=pre;
    pre=root.val;
    traversal(root.left);
  }
  traversal(root);
  return root;
};

File: 卡码/代码随想录/1/06-二叉树/Python/04-二叉搜索树属性/LeetCode_0538.py
#538. 把二叉搜索树转换为累加树
from typing import *
from ..TreeNode import *

class Solution:
  def __init__(self):
    self.pre:int=0
  def traversal(self,root:Optional[TreeNode]):
    if not root:
      return None
    self.traversal(root.right)
    root.val+=self.pre
    self.pre=root.val
    self.traversal(root.left)
  def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
    self.pre=0
    self.traversal(root)
    return root
